Add FML feature documentation for additional per feature metadata (#517)
mozilla/experimenter-docs: docs/deep-dives/specifications/fml/feature-metadata.mdx

---
id: feature-metadata
title: Providing feature metadata
slug: /fml/feature-metadata
sidebar_position: 7
---

> The feature manifest defines "features" which are used to

Reviewer: ```suggestion
The feature manifest defines "features" which are used to:
```

- provide default values for the configuration when not specified remotely by Nimbus
- provide experiment owners and others insight into the feature via experimenter, the Nimbus web site.

This last item is enabled by providing metadata about each feature in the feature manifest.

## A complete example

```yaml
    description: |
      The in-app messaging system.

    documentation:
      - name: User documentation
        url: https://experimenter.info/messaging/mobile-messaging
      - name: QA documentation
        url: https://docs.google.com/document/d/555-555/edit?usp=sharing

    meta-bug: https://bugzilla.mozilla.org/show_bug.cgi?id=5555555

    contacts:
      - jhugman@mozilla.com
      - amejiamarmol@mozilla.com
      - twhite@mozilla.com
      - brennie@mozilla.com

    events:
      - https://dictionary.telemetry.mozilla.org/apps/fenix/metrics/messaging_message_shown
      - https://dictionary.telemetry.mozilla.org/apps/fenix/metrics/messaging_message_clicked
      - https://dictionary.telemetry.mozilla.org/apps/fenix/metrics/messaging_message_dismissed
      - https://dictionary.telemetry.mozilla.org/apps/fenix/metrics/messaging_message_expired

>       - https://dictionary.telemetry.mozilla.org/apps/fenix/metrics/messaging_malformed

Reviewer 2: 🤩

    configurator: https://mozilla.github.io/limelight/

```

## `description`

This is a short description that appears alongside the name in experimenter.

It also is used to generate in-line class documentation in generated code.

## `documentation`

This is a list of named links to documentation for the feature elsewhere.

> Docummentation is might be for experiment owners to read, engineers to read, or QA.

Reviewer: ```suggestion
Documentation might be for experiment owners to read, engineers to read, or QA.
```

> Documentation for each variable and field is asked for in the feature manifest; for simpler features, this may be sufficient as experiment owner documentation,however, for more complex features a more cohesive document for the less technical audience may be more appropriate.

Reviewer: ```suggestion
Documentation for each variable and field is asked for in the feature manifest; for simpler features, this may be sufficient as experiment owner documentation, however, for more complex features a more cohesive document for a less technical audience may be more appropriate.
```
